test/middleware/consent.test.js

```diff
@@ -0,0 +1,73 @@
+/*global describe, it, beforeEach*/
+const request = require('supertest');
+const nextExpress = require('../../main');
+const expect = require('chai').expect;
+
+describe('Consent middleware', function () {
+	let app;
+	let locals;
+
+	before(function () {
+		app = nextExpress({
+			systemCode: 'consent',
+			withConsent: true
+		});
+		app.get('/', function (req, res) {
+			locals = res.locals;
+			res.sendStatus(200).end();
+		});
+	});
+
+	it('Should set the res.locals.consent property', function (done) {
+		request(app)
+			.get('/')
+			.expect(function () {
+				expect(locals).to.have.property('consent');
+			})
+			.end(done);
+	});
+
+	it('Should handle the FT-Consent="-" header', function (done) {
+		request(app)
+			.get('/')
+			.set('FT-Consent', '-')
+			.expect(function () {
+				expect(locals.consent).to.be.empty.and.an('object');
+			})
+			.end(done);
+	});
+
+	context('Should set the res.locals.consent property based on the FT-Consent header', () => {
+		[
+			{
+				header: 'marketingByemail:on,recommendedcontentOnsite:off',
+				expectedObject: {
+					marketingByemail: true,
```

Comment:
strange camel-casing on both these keys, is it correct?

Reply:
Yes, correct – convention is set elsewhere: https://github.com/Financial-Times/n-consent-proxy-client/pull/4/files#diff-04c6e90faac2675aa89e2176d2eec7d8R79